src/app/client/src/app/modules/learn/components/course-consumption/course-player/course-player.component.ts

```diff
@@ -434,9 +437,45 @@ export class CoursePlayerComponent implements OnInit, OnDestroy {
       this.router.navigate(['/learn/course/play', collectionUnit.identifier], navigationExtras);
     } else {
       this.showJoinTrainingModal = true;
+      const currentDate = new Date();
+      if(this.courseHierarchy.batches && this.courseHierarchy.batches.length === 1){
```

Comment:
I feel this is not the optimized code and also if startDate is not there then the js will fail